Enhancement/140 Bump WordPress and PHP minimums

faisal-alvi opened the pull request:
### Description of the Change
Bumped the minimum WordPress and PHP minimums.
* WordPress to 5.7
* PHP to 7.4

<!-- Enter any applicable Issue number(s) here that will be closed/resolved by this PR. -->
Closes #140.

### How to test the Change
* Check if the plugin is working as expected.
* Check the Github Actions working without fails.

### Changelog Entry
> Changed - Bump min WordPress to 5.7 and min PHP to 7.4

### Credits
<!-- Please list any and all contributors on this PR so that they can be added to this projects CREDITS.md file. -->
Props @vikrampm1 @faisal-alvi @cadic 


### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you are unsure about any of these, please ask for clarification.  We are here to help! -->
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

faisal-alvi committed:
enhancement/140 Bump WordPress and PHP minimums

faisal-alvi committed:
fix for deprecated schema:
Warning:       Your XML configuration validates against a deprecated schema.
Suggestion:    Migrate your XML configuration using "--migrate-configuration"!

faisal-alvi committed:
trying to fix PHPUnit version compatibility issue
> PHPUnit 9.5.21 #StandWithUkraine
> EEEEEEEEEEEEEEEEEEEEEEEEEEEEEE   30 / 30 (100%)

faisal-alvi committed:
Downgrading min php version to fix phpunit test failures
 - Root composer.json requires php >=7.4 but your php version (7.1.33) does not satisfy that requirement.

faisal-alvi commented:
The PHP Unit tests are failing with PHPUnit `9.5.21` version. Hence I have changed the min required PHP in the composer file to `7.1`, which keeps PHPUnit `7.5.20` and the tests run successfully.
```
"require": {
    "php": ">=7.1"
  },
```
@cadic I saw you mentioned about this somewhere recently, any suggestions here would be helpful.

cadic commented on composer.json:
```diff
@@ -19,7 +19,7 @@
   },
   "require": {
-    "php": ">=5.6"
+    "php": ">=7.1"
```
Should we put 7.4 here?

faisal-alvi replied:
I had put `7.4` but as mentioned in the [comment above](https://github.com/10up/simple-local-avatars/pull/143#issuecomment-1192505559), it upgrades the PHPUnit to `9.5.21` and all tests [failed](https://github.com/10up/simple-local-avatars/runs/7467741053?check_suite_focus=true). Hence I reverted it back to `7.1` (https://github.com/10up/simple-local-avatars/pull/143/commits/0712ba02be6a580f8016e2fadf4b7d12f70cd7df) that fixed this issue. Any suggestions?

faisal-alvi replied:
I noticed that the PHPUnit failure was due to the fix required in the upstream and the PR is already raised https://github.com/10up/wp_mock/pull/164.

Upgraded to `7.4` (https://github.com/10up/simple-local-avatars/pull/143/commits/c46a7337186d93b34c21358400f48b70fd8e8a5c)

cadic replied:
getAnnotations() error is fixed by changing `parent::setUp()` to `\WP_Mock::setUp()` in the test declarations https://github.com/10up/simple-local-avatars/pull/143/commits/2b8e16054e6cf9322728cd2fe8aaff0de04d7231

But there are still 3 tests fail after this fix
> Tests: 30, Assertions: 270, Errors: 1, Failures: 2.

cadic commented on .github/workflows/test.yml:
```diff
@@ -31,9 +31,6 @@ jobs:
-        # We claim to support from 5.6+ but WP Mock only supports 7.1+
-        # Also an issue with WP Mock not working on PHPUnit 9.5+, which PHP 7.3+ needs.
-        # php: [ '5.6', '7.0', '7.4', '8.0', '8.1' ]
         php: [ '7.1', '7.2' ]
```
7.1 and 7.2 looks like a deprecated support, needs to be updated to 7.4 and 8.0?

faisal-alvi replied:
Yes, we should, but need to fix the PHPUnit test failure issue first as mentioned above.

faisal-alvi replied:
Done, https://github.com/10up/simple-local-avatars/pull/143/commits/c46a7337186d93b34c21358400f48b70fd8e8a5c

cadic reviewed (changes requested):
@faisal-alvi thank you for this update, I left couple questions about PHP versions which are still below the current minimum.

faisal-alvi committed:
upgrading to `7.4` after noticing the PHPUnit failure is already addressed in the upstream:
https://github.com/10up/wp_mock/pull/164

faisal-alvi commented:
As mentioned in https://github.com/10up/simple-local-avatars/pull/143#discussion_r942310144, the PHPUnit failure is due to the fix required in the upstream, and the PR is already raised https://github.com/10up/wp_mock/pull/164, so we can ignore the PHPUnit failures in the Github action(s).

cadic committed:
Fix all tests failure

cadic reviewed (changes requested):
Please check 3 tests are still failing after fixing the getAnnotation error

faisal-alvi committed:
fix phpunit test failures after fixing the getAnnotation error

faisal-alvi committed:
trying to fix:
1) SimpleLocalAvatarsNetworkTest::test_is_network
Error: Call to undefined method SimpleLocalAvatarsNetworkTest::getAnnotations()
https://github.com/10up/simple-local-avatars/runs/7919270614?check_suite_focus=true

faisal-alvi committed:
fix `getAnnotations` error for multisite tests
https://github.com/10up/simple-local-avatars/runs/7919310499?check_suite_focus=true

cadic reviewed (approved):
LGTM!
